gdnative-log/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+//! # Adapter for the Rust log crate
+//!
+//! This crate writes an adapter for the Rust log crate such that logging statements
+//! get output on the Godot console. Essentially it translates `error!()`, `info!()`
+//! and so on into the appropriate `godot_error!()`, `godot_warn!()` and
+//! `godot_print()` calls.
+
+#[macro_use]
+extern crate gdnative_core;
+
+use log::{self, Level, Metadata, Record, SetLoggerError};
+
+struct GodotLogger {
+    level: Level,
+}
+
+impl log::Log for GodotLogger {
+    fn enabled(&self, metadata: &Metadata) -> bool {
+        metadata.level() <= self.level
+    }
+
+    fn log(&self, record: &Record) {
+        if self.enabled(record.metadata()) {
+            match record.level() {
+                Level::Warn => godot_warn!("{} - {}", record.level(), record.args()),
+                Level::Error => godot_error!("{} - {}", record.level(), record.args()),
+                _ => godot_print!("{} - {}", record.level(), record.args()),
+            }
+        }
+    }
+
+    fn flush(&self) {}
+}
+
+/// Initialises the logger with the warning level set.
+///
+/// ```ignore
+/// # #[macro_use] extern crate log;
+/// #
+/// # fn main() {
+/// gdnative::init_logging_with_level(log::Level::Warn).unwrap();
+///
+/// warn!("This will be logged as a warning.");
+/// info!("This message will not be logged.");
+/// # }
+/// ```
+pub fn init_logging_with_level(level: Level) -> Result<(), SetLoggerError> {
```

Comment:
Nitpick: publicly reexporting the `Level` type here would allow users to call `init_logging_with_level` without having to explicit depend on the log crate.

Reply:
Good point, fixed